.github/workflows/make_prs_for_other_repos.yaml

```diff
@@ -1,48 +1,74 @@
 name: "Make PRs for Nextstrain projects which depend on Auspice"
 on:
-  # pull_request:
-  push:
-    branches: 
-      - "not-a-valid-branch"
+  workflow_dispatch:
 jobs:
   make-pr-on-nextstrain-dot-org: # <job_id>
+    # I don't see this being used for tags, so ensure it's only run on branches
+    # to make subsequent logic and wording easier.
+    if: github.ref_type == 'branch'
+
     runs-on: ubuntu-latest
+
+    env:
+      DESTINATION_REPO_DIR: nextstrain.org
+
     steps:
+      # Outputs:
+      # - pr-number: The PR number from the branch name (exits if no PR exists).
+      # - auspice-sha: The GitHub-managed merge ref. Used for npm install.
+      - name: Detect PR from branch
+        id: detect-pr
+        run: |
+          PR_NUMBER=$(gh pr view $GITHUB_REF_NAME --repo nextstrain/auspice --json 'number' --jq '.number') || true
+          if [[ -z $PR_NUMBER ]]; then
+            echo "ERROR: This branch is not associated with a PR in Auspice." >&2
+            exit 1
+          fi
+          MERGE_SHA=$(gh pr view $GITHUB_REF_NAME --repo nextstrain/auspice --json 'potentialMergeCommit' --jq '.potentialMergeCommit.oid')
```

Comment:
Just double checking I understand what's going on here (via https://github.com/nextstrain/auspice/pull/1565#issuecomment-1269040879) -- the salient thing is that it's a `workflow_dispatch` trigger and so `$GITHUB_SHA` here refers to the head of the branch (which in this case is a PR, but that's not relevant for how the action machinery operates). However, if the action _had_ been triggered by a push-to-branch then `$GITHUB_SHA` would refer to the merge commit SHA?

Perhaps add a one-line comment here explaining?